src/nun.js

```diff
@@ -1,35 +1,39 @@
 import NunDb from 'nun-db';
-import { syncGameState } from './redux-flow/reducers/tic-tac-toe/action-creators';
-import { TIC_TAC_TOE_PREFIX } from './redux-flow/reducers/tic-tac-toe/actions';
+import { syncGameState, join } from './redux-flow/reducers/tic-tac-toe/action-creators';
+import { TIC_TAC_TOE_PREFIX, JOIN_ROOM } from './redux-flow/reducers/tic-tac-toe/actions';
 
 const nun = new NunDb('wss://ws.nundb.org', 'tic-tac-toe', 'tic-tac-toe12i3ukjsd');
 
-let ignore = false;
-const dbMiddleware = (store) => {
-  nun.watch('lastEvent', (action) => {
-    const ticTacToeAction = new RegExp(`^${TIC_TAC_TOE_PREFIX}`);
-    ignore = true;
-    if (ticTacToeAction.test(action.value.type)) { store.dispatch(action.value); }
-  });
+const ticTacToeAction = new RegExp(`^${TIC_TAC_TOE_PREFIX}`);
+const dbMiddleware = (store) => (next) => (action) => {
+  next(action);
```

Comment:
Important to call next before the rest of this function (Or it would sync the state before applying the current event)